pkg/frontend/kubeactions/kubeactions.go

```diff
@@ -160,6 +163,63 @@ func (ka *kubeactions) List(ctx context.Context, oc *api.OpenShiftCluster, kind,
 	return ul.MarshalJSON()
 }
 
+func (ka *kubeactions) applyOne(ctx context.Context, dyn dynamic.Interface, grs []*restmapper.APIGroupResources, un *unstructured.Unstructured) error {
+	if strings.EqualFold(un.GetKind(), "secret") { // TODO do we need this?
+		return fmt.Errorf("access to secrets is forbidden")
+	}
+	gvrs := ka.findGVR(grs, un.GetKind())
+	if len(gvrs) == 0 {
+		return fmt.Errorf("Kind %s not found", un.GetKind())
+	}
+	if len(gvrs) > 1 {
+		return fmt.Errorf("Kind %s did not uniquely match one GRV", un.GetKind())
+	}
+	gvr := gvrs[0]
+
+	ka.log.Debugf("appling kube object %s %s %s ", un.GetKind(), un.GetNamespace(), un.GetName())
```

Comment:
I think we should log the current and the new state of the object.

Reply:
when you say "state" do you mean the whole object? If so I'd worry about secrets being in the env section. I'll have a look and see what sync pod does..